app/uk/gov/hmrc/merchandiseinbaggage/controllers/TimeOutController.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.mvc.MessagesControllerComponents
+import uk.gov.hmrc.merchandiseinbaggage.config.AppConfig
+import uk.gov.hmrc.merchandiseinbaggage.repositories.DeclarationJourneyRepository
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class TimeOutController @Inject()(override val controllerComponents: MessagesControllerComponents,
+                                  actionProvider: DeclarationJourneyActionProvider,
+                                  repo: DeclarationJourneyRepository,
+                               )(implicit ec: ExecutionContext, appConfig: AppConfig)
+  extends DeclarationJourneyController {
+
+  override val onPageLoad = actionProvider.journeyAction.async { implicit request =>
```

Comment:
would this endpoint be triggered several times during the timeout popup or only once ?  
if its called several times, not sure if we need to upsert that many times - 
ofcourse in any-case not a big deal considering our production traffic :)

Reply:
should be only once